paramiko/client.py

```diff
@@ -228,6 +228,7 @@ def connect(
         gss_host=None,
         banner_timeout=None,
         auth_timeout=None,
+        gss_trust_dns=True
```

Comment:
This should probably be next to the other gss args - I think it's pretty unlikely that anyone is depending on the args order instead of using kwargs by midpoint of this args list.

I also wonder if most of the gss args can be boiled down to a single "gss context" argument. (For backwards compat it would still need an additional kwarg here, but it would be the last one for gss ;)

Reply:
I think for now the (yes, minor) benefit of preserving backwards compat arg order is worth leaving this alone, but partly _because_ I agree that later we should probably compress them into a single argument. (Counterpoint: doing that makes it harder to do things like create partial applications of the function...but I'm gonna guess not many people are doing that.)